portfolio/src/main/java/com/google/sps/servlets/ListQuestionsServlet.java

```diff
@@ -24,12 +24,20 @@
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String limit = request.getQueryString();
```

Comment:
I think it would be more robust to parse this into an int, and set a default if parsing fails.

(e.g. what if I call this with a query string like "?xyz" ?

Reply:
In terms of performance, I am not sure that using String is better but because in the if condition I cast the counter into a string and then compare, I don't think it would cause a problem if we call this with a query string like "?xyz". On the other hand, if we use an integer, we have to make sure that the default value is not a value we can have in our datastore!

Reply:
you're comparing a counter to a limit in line 36. 

These are conceptually integers and their type should reflect that.

I'm not sure why you need to check that it's not in the datastore. You're not comparing against datastore entries, but rather against the counter.

Reply:
True! I was wrong, it has nothing to do with the datastore. I don't know why I had in mind that this the id we"re talking about. I will change this right away. Thanks for the feedback Erick!